Gateway/MinistryPlatform.Translation/Repositories/FormSubmissionRepository.cs

```
@@ -160,15 +162,29 @@ private static IDbCommand CreateTripFormResponsesSqlCommandWithSelectionId(int s
 
         public int SubmitFormResponse(MpFormResponse form)
         {
-            var token = ApiLogin();
-            var responseId = CreateOrUpdateFormResponse(form, token);
-            foreach (var answer in form.FormAnswers)
+            try
             {
-                if (answer.Response == null) continue;
-                answer.FormResponseId = responseId;
-                CreateOrUpdateFormAnswer(answer, token);
+                var token = ApiLogin();
+                var responseId = CreateOrUpdateFormResponse(form, token);
+                foreach (var answer in form.FormAnswers)
+                {
+                    if (answer.Response == null) continue;
+                    answer.FormResponseId = responseId;
+                    CreateOrUpdateFormAnswer(answer, token);
+                }
+                return responseId;
             }
-            return responseId;
+            catch
+            {
+                JsonSerializerSettings settings = new JsonSerializerSettings
+                {
+                    Error = (serializer, err) => err.ErrorContext.Handled = true
+                };
+                string json = JsonConvert.SerializeObject(form, settings);            
+                _logger.Error($"Failure in SubmitFormResponse: {json}");
+                throw new Exception();
```

Should we re-throw the original exception vs. throwing a new exception ?

---

Yes!